octree_web_viewer/src/bin/points_web_viewer.rs

```diff
@@ -12,92 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use actix_web::http::Method;
-use actix_web::{server, HttpRequest, HttpResponse};
-use clap::value_t;
-use octree_web_viewer::backend::{NodesData, VisibleNodes};
+use octree_web_viewer::utils;
 use point_viewer::octree;
 use std::path::PathBuf;
 use std::sync::Arc;
-
-const INDEX_HTML: &str = include_str!("../../client/index.html");
-const APP_BUNDLE: &str = include_str!("../../../target/app_bundle.js");
-const APP_BUNDLE_MAP: &str = include_str!("../../../target/app_bundle.js.map");
-
-const DEFAULT_PORT: &str = "5433";
-
-fn index(_req: &HttpRequest) -> HttpResponse {
-    HttpResponse::Ok()
-        .content_type("text/html")
-        .body(INDEX_HTML)
-}
-
-fn app_bundle(_req: &HttpRequest) -> HttpResponse {
-    HttpResponse::Ok()
-        .content_type("text/html")
-        .body(APP_BUNDLE)
-}
-
-fn app_bundle_source_map(_req: &HttpRequest) -> HttpResponse {
-    HttpResponse::Ok()
-        .content_type("text/html")
-        .body(APP_BUNDLE_MAP)
+use structopt::StructOpt;
+
+/// HTTP web viewer for 3d points stored in OnDiskOctrees
+#[derive(StructOpt, Debug)]
+#[structopt(name = "points_web_viewer", about = "Visualizing points")]
+struct CommandlineArguments {
+    /// The octree directory to serve, including a trailing slash.
+    octree_path: String,
+    /// Port to listen on.
+    #[structopt(default_value = "5433", long = "--port")]
+    port: u16,
+    /// IP string.
+    #[structopt(default_value = "127.0.0.1", long = "--ip")]
+    ip: String,
 }
 
 fn main() {
-    //TODO(catevita): Convert this to structopt
-    let matches = clap::App::new("octree_web_viewer")
-        .args(&[
-            clap::Arg::with_name("port")
-                .help("Port to listen on for connections.")
-                .long("port")
-                .default_value(DEFAULT_PORT)
-                .takes_value(true),
-            clap::Arg::with_name("octree_directory")
-                .help("Input directory of the octree directory to serve.")
-                .index(1)
-                .required(true),
-        ])
-        .get_matches();
+    let args = CommandlineArguments::from_args();
+    println!("{:?}", args);
 
-    let port = value_t!(matches, "port", u16).unwrap();
-    let ip_port = format!("127.0.0.1:{}", port);
-    let octree_directory = PathBuf::from(matches.value_of("octree_directory").unwrap());
+    let ip_port = format!("{}:{}", args.ip, args.port);
 
+    let octree_directory = PathBuf::from(&args.octree_path);
     // The actix-web framework handles requests asynchronously using actors. If we need multi-threaded
     // write access to the Octree, instead of using an RwLock we should use the actor system.
+
+    // TODO(catevita) octree factory
+
```

Comment:
Nit: Remove empty line.

Reply:
I will take care of it in the octree factory PR